RationaleMCP/0035/Readme.md

```diff
@@ -0,0 +1,129 @@
+* Support cases might occur less often, due to better understanding of the description. 
+* A new translation is additionally a review of the current state and can result in a better quality of both language versions.
+* Users can even write their own translation, when having a template *.pot-File for the library. 
+
+The use case for the proposal is the MSL 4.0. 
+* POT-File of current MSL 4.0 library --> this translation template is provided and filled for the Hydraulics Pump. 
+(*add pictures*)
+
+Proposed changes to the specification are described in SpecificationText.md. It consists of a new section **13.6 Multilingual Descriptions** which contains:
+* Where the translation files to be stored
+* How does a GNU gettext translation template for a library look like and which kind of descriptive texts shall be included?
+* Which strings should be translated:
+
+The precise updated text of the specification is part of this branch/pull-request: Yes.
+
+# Backwards Compatibility
+The proposal is backwards compatible. All multilingual text changes can be applied without effect at existing Modelica code. 
+After the introduction of translation files for libraries, this files need to be adjusted with every library change which adds new commented parts or changes existing comments.
+
+# Tool Implementation
+
+## Experience with Prototype
+### Implementation steps
+#### 1. Read out a translation template file
+Create at first a translation template file \*.pot for a Modelica library. Means a (tool dependent) readout of all needed text strings into a *LibraryName*.pot-File. It contains: 
+- a header which concludes all necessary information
+![Header of a *.pot file](POT_Header.png)
+
+- for all original strings entries with following structure: 
+```
+    msgctxt "Hydraulics.Basics.TankOverflow"
+    msgid "Volume Flow Port A"
+    msgstr ""
+```
+
+Where *msgctxt* contains the name of the Modelica Class and *msgid* contains the textstring which is situated in this name space and shall be translated. Please regard that if a *msgid* string (in this example "Volume Flow Port A") is given more than once in a namespace, all occurrences are translated with the same translation!
+
+It shall be mandatory to use for *msgctxt* the full name of a class:
+- so a specific translation depending on the content in this namespace is possible, as every textstring may differ depending on the context,
+- using the class makes it easier for the Modelica tool to copy and re-arrange models and packages without loosing the already existing language information.
+
+The following Modelica constructs shall be read out / translated: 
+* description strings
+* strings in following annotations:
+  * Text.string, Text.textString
+  * missingInnerMessage, obsolete, unassignedMessage 
+  * Dialog.[group|tab] 
+  * Dialog.[load|save]Selector.[caption|filter] 
+  * Documentation.[info|revisions]
+  * Figure.title, Plot.title, Curve.legend

```

Comment:
Few constructs are missing e.g., `mustBeConnected`, `Figure.group`, `Figure.caption` and `Plot.label`. There might be more.

Reply:
Good catch.  I think you mean `Axis.label` rather than `Plot.label`.  Also missing: `Curve.legend`.

Reply:
Again, I realize this is not the actual specification text, so it doesn't really matter that the list is incomplete.  Let's continue at the right place instead: https://github.com/modelica/ModelicaSpecification/pull/2956/files#r1011752327